backend/app/routers/metadata.py

```diff
@@ -70,6 +71,49 @@ async def get_metadata_definition_list(
     return page.dict()
 
 
+@router.put(
+    "/definition/{metadata_definition_id}", response_model=MetadataDefinitionOut
+)
+async def update_metadata_definition(
+    metadata_definition: MetadataDefinitionIn,
+    metadata_definition_id: str,
+    user=Depends(get_current_user),
+):
+    existing_count = await MetadataDefinitionDB.find(
+        {
+            "$and": [
+                {"_id": {"$ne": PydanticObjectId(metadata_definition_id)}},
+                {"name": {"$eq": metadata_definition.name}},
+            ]
+        }
+    ).count()
+    if existing_count > 0:
+        raise HTTPException(
+            status_code=409,
+            detail=f"Metadata definition named {metadata_definition.name} already exists.",
```

Comment:
Shouldn't this be the opposite? We would want to update the existing metadata.

Reply:
This is for the scenario where the users update the name and the updated name is same as some other existing metadata definition name

Reply:
My understanding is - we want to keep the metadata definiton name as unique. Metadata definition name is used to connect definition to actual metadata.